go-controller/pkg/ovn/master.go

```diff
@@ -397,7 +400,16 @@ func addStaticRouteToHost(node *kapi.Node, nicIP string) error {
 func parseNodeHostSubnet(node *kapi.Node) (*net.IPNet, error) {
 	sub, ok := node.Annotations[OvnHostSubnet]
 	if !ok {
-		return nil, fmt.Errorf("Error in obtaining host subnet for node %q for deletion", node.Name)
+		sub, ok = node.Annotations[OvnHostSubnetLegacy]
+	} else {
+		nodeSubnets := make(map[string]string)
+		if err := json.Unmarshal([]byte(sub), &nodeSubnets); err != nil {
+			return nil, err
+		}
+		sub, ok = nodeSubnets["default"]
+	}
+	if !ok {
+		return nil, fmt.Errorf("error in obtaining host subnet for node %q", node.Name)
```

Comment:
it's more like "node %q has no subnet annotation"

Reply:
Fixed